File: OffByOne/src/OffByOne.Qls/Ast/Style/Widgets/SpinboxWidget.cs
namespace OffByOne.Qls.Ast.Style.Widgets
{
    using OffByOne.Qls.Ast.Style.Widgets.Base;
    using OffByOne.Qls.Common.Visitors.Contracts;

    public class SpinboxWidget : Widget
    {
        public override TResult Accept<TResult, TContext>(
            IWidigetVisitor<TResult, TContext> visitor,
            TContext environment)
        {
            return visitor.Visit(this, environment);
        }
    }
}

File: AlternativeQL/src/org/qls/ast/widget/SpinboxWidget.java
package org.qls.ast.widget;

import org.ql.ast.type.*;

public class SpinboxWidget extends Widget {

    public boolean isCompatibleWith(FloatType type) {
        return true;
    }

    public boolean isCompatibleWith(IntegerType type) {
        return true;
    }

    public boolean isCompatibleWith(MoneyType type) {
        return true;
    }
}
